sandbox/plc4c/examples/hello-subsciptions/src/hello_subscriptions.c

```diff
@@ -0,0 +1,264 @@
+
+enum plc4c_connection_state_t {
+  CONNECTING,
+  CONNECTED,
+  SUBSCRIPTION_REQUEST_SENT,
+  SUBSCRIPTION_RESPONSE_RECEIVED,
+  READING_EVENTS,
+  DISCONNECTING,
+  DISCONNECTED
+};
+typedef enum plc4c_connection_state_t plc4c_connection_state;
+
+#pragma clang diagnostic push
+#pragma ide diagnostic ignored "hicpp-multiway-paths-covered"
+
+int main() {
+  bool loop = true;
+  plc4c_system *system = NULL;
+  plc4c_connection *connection = NULL;
+  plc4c_subscription_request *subscription_request = NULL;
+  plc4c_subscription_request_execution *subscription_request_execution = NULL;
+  int num_events = 0;
+  void *subscription_handle = NULL;
+
+  // Create a new uninitialized plc4c_system
+  printf("Creating new PLC4C System (Initializing inner data-structures) ... ");
+  plc4c_return_code result = plc4c_system_create(&system);
+  if (result != OK) {
+    printf("FAILED\n");
+    return -1;
+  }
+  printf("SUCCESS\n");
+
+  // Manually register the "simulated" driver with the system.
+  printf("Registering driver for the 'simulated' protocol ... ");
+  plc4c_driver *simulated_driver = plc4c_driver_simulated_create();
+  result = plc4c_system_add_driver(system, simulated_driver);
+  if (result != OK) {
+    printf("FAILED\n");
+    return -1;
+  }
+  printf("SUCCESS\n");
+
+  printf("Registering transport for the 'dummy' transport ... ");
+  plc4c_transport *dummy_transport = plc4c_transport_dummy_create();
+  result = plc4c_system_add_transport(system, dummy_transport);
+  if (result != OK) {
+    printf("FAILED\n");
+    return -1;
+  }
+  printf("SUCCESS\n");
+
+  // Initialize the plc4c_system (loading of drivers, setting up other stuff,
+  // ...)
+  printf(
+      "Initializing the PLC4C system (Loading of drivers and transports) ... ");
+  result = plc4c_system_init(system);
+  if (result != OK) {
+    printf("FAILED\n");
+    return -1;
+  }
+  printf("SUCCESS\n");
+
+  // Establish connections to remote devices
+  // you may or may not care about the connection handle
+  printf("Connecting to 'simulated://foo' ... ");
+  result = plc4c_system_connect(system, "simulated://foo", &connection);
+  if (result != OK) {
+    printf("FAILED\n");
+    return -1;
+  }
+
+  // Central program loop ...
+  plc4c_connection_state state = CONNECTING;
+  while (loop) {
+    printf("* ");
+
+    // Give plc4c a chance to do something.
+    // This is where all I/O is being done.
+    if (plc4c_system_loop(system) != OK) {
+      printf("ERROR in the system loop\n");
+      break;
+    }
+
+    // Depending on the current state, implement some logic.
+    switch (state) {
+      case CONNECTING: {
+        // Check if the connection is established:
+        if (plc4c_connection_get_connected(connection)) {
+          printf("SUCCESS\n");
+          state = CONNECTED;
+        } else if (plc4c_connection_has_error(connection)) {
+          printf("FAILED\n");
+          return -1;
+        }
+        break;
+      }
+      case CONNECTED: {
```

Comment:
I think you want to update the comments and explain the subscription stuff you are doing